circuit/pallets/execution-delivery/src/side_effect/mod.rs

```diff
@@ -0,0 +1,34 @@
+use codec::{Decode, Encode};
+use sp_runtime::RuntimeDebug;
+use sp_std::vec::Vec;
+
+pub type SideEffectId<T> = <T as frame_system::Config>::Hash;
+pub type TargetId = bp_runtime::ChainId;
+type Bytes = Vec<u8>;
+
+#[derive(Clone, Eq, PartialEq, Default, Encode, Decode, RuntimeDebug)]
+pub struct InboundSideEffect<AccountId, BlockNumber, BalanceOf> {
+    pub target: TargetId,
+    pub prize: BalanceOf,
+    pub ordered_at: BlockNumber,
+    pub encoded_action: Bytes,
+    pub encoded_args: Vec<Bytes>,
+    pub signature: Bytes,
+    pub enforce_executioner: AccountId,
+}
+
+#[derive(Clone, Eq, PartialEq, Default, Encode, Decode, RuntimeDebug)]
+pub struct OutboundSideEffect<AccountId, BlockNumber, BalanceOf> {
+    pub err: Option<Bytes>,
+    pub output: Option<Bytes>,
+    pub inclusion_proof: Option<Bytes>,
```

Comment:
I left inclusion_proof Optional since on Eth we'd rely on Snowfork's intrinsic inclusion proof for events. Could remove Option since could just be empty bytes